File: api/models/Report.js
const mongoose = require("mongoose");

const reportSchema = new mongoose.Schema({
  reporterId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "User",
    required: true,
  },
  reason: {
    type: String,
    required: true,
    trim: true,
  },
  documentId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "Document",
    required: true,
  },
  action: {
    type: String,
    default: "", 
    enum: [
      "",
      "Đã duyệt, document sẽ bị xóa sau 1 ngày",
      "Đã từ chối report",
      "Tài liệu đã bị xóa"
    ],
  },
  createDate: {
    type: Date,
    default: Date.now,
  },
});

module.exports = mongoose.model("Report", reportSchema);

File: app/lib/models/report.dart
class Report {
  String? id;
  String? reporterId;
  String? reason;
  String? documentId;
  String? action;
  String? createDate;

  Report({
    this.id,
    this.reporterId,
    this.reason,
    this.documentId,
    this.action,
    this.createDate,
  });

  Report.fromJson(Map<String, dynamic> json) {
    id = json['_id'];
    reporterId = json['reporterId'];
    reason = json['reason'];
    documentId = json['documentId'];
    action = json['action'];
    createDate = json['createDate'];
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'reporterId': reporterId,
      'reason': reason,
      'documentId': documentId,
      'action': action,
      'createDate': createDate,
    };
  }
}
